test/cpp/tensorexpr/test_schedule.cpp

```diff
@@ -545,5 +545,67 @@ void testScheduleDynamicShape2D() {
   testWithSize(37, 11);
 }
 
+void testLoopNest() {
+
+  KernelScope kernel_scope;
+  const int kVectorSize = 8;
+  const int kVectorCount = 128;
+  const int kSize1 = 1024;
+  const int kSize2 = 256;
+
+  VarHandle n("N", kHandle);
+  Buffer a(VarHandle("A", kHandle), kFloat, {n, ExprHandle(kSize1), ExprHandle(kSize2)});
+  Buffer b(VarHandle("B", kHandle), kFloat, {n, ExprHandle(kSize1), ExprHandle(kSize2)});
+  Buffer c(VarHandle("C", kHandle), kFloat, {n, ExprHandle(kSize1), ExprHandle(kSize2)});
+  Buffer d(VarHandle("D", kHandle), kFloat, {n, ExprHandle(kSize1), ExprHandle(kSize2)});
+
+  Tensor* e = Compute(
+      "e",
+      {{n, "n"}, {kSize1, "i"}, {kSize2, "j"}},
+      [&](const VarHandle& n, const VarHandle& i, const VarHandle& j) {
+        return a(n, i, j) + b(n, i, j);
+      });
+  Tensor* f = Compute(
+      "f",
+      {{n, "n"}, {kSize1, "i"}, {kSize2, "j"}},
+      [&](const VarHandle& n, const VarHandle& i, const VarHandle& j) {
+        return (*e)(n, i, j) + c(n, i, j);
+      });
+  Tensor* g = Compute(
+      "g",
+      {{n, "n"}, {kSize1, "i"}, {kSize2, "j"}},
+      [&](const VarHandle& n, const VarHandle& i, const VarHandle& j) {
+        return (*f)(n, i, j) + d(n, i, j);
+      });
+
+
+  // NEW API:
+  {
+    LoopNest l({e, f, g});
+    l.ComputeInline(l.getLoopBodyFor(e));
```

Comment:
Can we still provide the per_tensor interface? :)

Reply:
I consciously want to drop it as it's not as well defined as per-loop API. Not well defined because after even a single transformation tensor-computation will be represented by several loops, and the transformation only applies to one of them. In the current implementation it's worked around by using variables assigned to loops, but I think there is no real need in having this. Notice that the new API is not more verbose even in a number of lines needed, but it's more consistent IMO.

Reply:
In the original code, we probably just want to expose the axis (each axis corresponds to loop) to user. In the future, we want to define the reduction axis. I feel this is still necessary to give user the flexibility to operate on axis instead of each loop. In the new code, user needs to be aware of exact underlying loops to define the transformation,